ratatui-core/src/layout/rect.rs

```diff
@@ -351,6 +353,35 @@ impl Rect {
         }
     }
 
+    /// Centers the rect horizontally. The constraint indicates the result rects width.
+    #[must_use]
+    pub fn center_horizontally(self, constraint: Constraint) -> Self {
```

Comment:
Bikeshedding the name. I think I prefer `centered_` over `center_` as an adjective generally suggests that a new item that is modified will be returned. `center` is ambiguously either a noun or a verb, which is problematic. It's difficult to intuit whether it modifies the existing item or returns a new item. Does this make sense? (It's possible that I'm wrong here if there's other existing code examples that would use this name like this. I think this is unlikely, but only at around 90% confidence).